### ISSUE TITLE
forwardport: Alerting v1 not triggering/sending notifications #31365 

### ISSUE THREAD

**Issue by slickwarren** [rancher/rancher#31522]
forwardport:  rancher/rancher#31365 for master-head 

the [tickets mentioned here](https://github.com/rancher/rancher/issues/31365#issuecomment-785480061) were related to this failure

**Comment by slickwarren** [comment]
not working on master-head (caa261959550c2f104a450e54d00bf4d49e17c9c).  
`alertmanager-cluster-monitoring` in crashloopbackoff with this error in logs 
```
level=warn ts=2021-03-04T20:29:50.666Z caller=cluster.go:228 component=cluster msg="failed to join cluster" err="1 error occurred:\n\t* Failed to resolve alertmanager-cluster-alerting-0.alertmanager-operated:9094: lookup alertmanager-cluster-alerting-0.alertmanager-operated on 10.43.0.10:53: no such host\n\n"
... 
level=error ts=2021-03-04T20:24:45.688Z caller=coordinator.go:124 component=configuration msg="Loading configuration file failed" file=/etc/alertmanager/config/alertmanager.yaml err="not a valid duration string: \"6d2h\""
```

**Comment by aiyengar2** [comment]
Is this still an issue? iirc the fix was not with the Monitoring chart, but rather with rancher/rancher changes.

**Comment by aiyengar2** [comment]
Was able to reproduce the issue. Seeing the following on master-head:

### /namespaces/cattle-prometheus/yaml is returning an invalid object

https://RANCHER_URL/v3/cluster/c-ck775/namespaces/cattle-prometheus/yaml

```json
{
"baseType": "error",
"code": "ServerError",
"message": "invalid character '\\x1f' looking for beginning of value",
"status": 500,
"type": "error"
}
```

This only happens with the `cattle-prometheus` namespace after Catalog V1 deploys `cluster-alerting` for Alerting V1.

### Alertmanager Config Secret has wrong input

Within the alertmanager-cluster-alerting secret (the Alertmanager Config for Monitoring V1), I also see:

```yaml
global:
  resolve_timeout: 5m
  smtp_require_tls: false
  slack_api_url: https://api.slack.com
  pagerduty_url: https://events.pagerduty.com/v2/enqueue
route:
  receiver: rancherlabs
  routes:
  - receiver: c-ck775:cag-mj4gp
    group_by:
    - group_id
    match:
      group_id: c-ck775:cag-mj4gp
    continue: true
    routes:
    - group_by:
      - rule_id
      - resource_kind
      - target_namespace
      - target_name
      - event_message
      match:
        rule_id: c-ck775:cag-mj4gp_car-rzcbn
      continue: true
      group_wait: 1s
      group_interval: 1s
      repeat_interval: 6d2h
    group_wait: 30s
    group_interval: 3m
    repeat_interval: 1h
  group_wait: 1m
  group_interval: 10s
  repeat_interval: 1h
receivers:
- name: rancherlabs
  slack_configs:
  - send_resolved: false
    channel: '#alert'
    title_link: ""
- name: c-ck775:cag-mj4gp
  slack_configs:
  - send_resolved: false
    api_url: <REDACTED by @aiyengar2>
    color: '{{ if eq (index .Alerts 0).Labels.severity "critical" }}danger{{ else
      if eq (index .Alerts 0).Labels.severity "warning" }}warning{{ else }}good{{
      end }}'
    title: '{{ template "rancher.title" . }}'
    title_link: ""
    text: '{{ template "slack.text" . }}'
templates:
- /etc/alertmanager/config/notification.tmpl
```

In the above configuration, I'm not sure where `repeat_interval: 6d2h` is coming from. 

#### Workload service generation controller is logging errors

Within the logs, I see the following, which means that this could be related to https://github.com/rancher/rancher/issues/32692:

```
2021/05/20 17:21:57 [ERROR] error syncing 'cattle-prometheus/prometheus-operator-monitoring-operator': handler workloadServiceGenerationController: cannot find app namespace in labels of cattle-prometheus, requeuing
2021/05/20 17:22:05 [ERROR] error syncing 'cattle-prometheus/alertmanager-cluster-alerting': handler workloadServiceGenerationController: cannot find app namespace in labels of cattle-prometheus, requeuing
```

**Comment by aiyengar2** [comment]
For future reference, the reason why Alerting V1 broke was due to a subtle breaking change in an underlying repository: prometheus/common. 

The underlying repository [changed the definition that was outputted by the `String()` function](https://github.com/prometheus/common/commit/9c5aa1eb995200ed442a9ca4d8b9694922d9cf93) on a `model.Duration` object in release v0.11.0+, and while [rancher@release/v2.5](https://github.com/rancher/rancher/blob/87006de30625c046523a329b03ea2580db35fb57/go.mod#L88) and upstream Alertmanager was still using prometheus/common v0.10.0, [rancher@master](https://github.com/rancher/rancher/blob/465bb3b80cb9f2d8c002fa9d7c37be20d38ed411/go.mod#L84) was bumped to v0.14.0, as noted in https://github.com/rancher/rancher/issues/31365#issuecomment-785847222.

[rancher@master](https://github.com/rancher/rancher/blob/d31a7feceeb30825bccac9dd45217771bd70fa21/go.mod#L96) has since been upgraded further due to an upgrade required to support k8s 1.21 and upstream Alertmanager has released v0.22.0 (as seen in https://github.com/prometheus/alertmanager/commit/470634d49fc629727d91cbb271d7c6e9bc282b06), which contains the version of prometheus/common that is >v0.10.0.

Therefore, the fix for this issue is to simply bump the Alertmanager version, but we will need to ensure that this fix is backported to all current Rancher releases (2.4, 2.5, 2.6), resulting in a brand-new chart for all of them.

**Comment by slickwarren** [comment]
with alerting v1 enabled, I tested the following:
* deploy alerting v1 on rancher v2.5.8 cluster -> setup alert + notifier -> upgrade to master-head (f90b36efb4cf50088e63e081e359ddd9b94c8ecf) -- pass
* deploy alerting v1 on rancher master-head (f90b36efb4cf50088e63e081e359ddd9b94c8ecf) -> setup alert + notifier -> pass 
* verified version of alerting is `0.22.0`
